pulp-glue/pulp_glue/common/openapi.py

```diff
@@ -44,6 +49,28 @@ class UnsafeCallError(OpenAPIError):
     pass
 
 
+class OpenAPISecurityScheme:
+    def __init__(self, security_scheme: SecurityScheme):
+        self.security_scheme = security_scheme
+
+        self.parse()
+
+    def parse(self) -> None:
+        self.type = self.security_scheme["type"]
+        self.description = self.security_scheme.get("description", "")
+
+        if self.type == "oauth2":
+            self.flows: OAuth2Flows = self.security_scheme.get("flows")
+            client_credentials: ClientCredentials = self.flows["clientCredentials"]
+            if client_credentials:
+                self.flow_type: t.Optional[str] = "clientCredentials"
+                self.token_url: str = client_credentials["tokenUrl"]
+                self.scopes: OAuth2FlowsScopes = list(client_credentials.get("scopes").keys())
+
+        if self.type == "http":
+            self.scheme = self.security_scheme["scheme"]
```

Comment:
This is all kind of hazy.
Can you explain why the logic didn't fit in the authproviderbase call method?

Reply:
Does the AuthProviderBase need to understand how to properly parse an OpenAPI SecurityScheme for each of those authentication methods? I was thinking on abstracting this to AuthProviderBase and exposes the minimum needed to choose which method to call.